include/fsm/fill_spill_merge.hpp

```diff
@@ -677,7 +686,8 @@ static SubtreeDepressionInfo FindDepressionsToFill(
   //water will have been transferred into the parent and we don't want to pool
   //the parent's water with our own (it might be at the bottom of a cliff).
 
-  if(this_dep.water_vol<this_dep.dep_vol || this_dep.ocean_parent){
+  if(this_dep.water_vol<this_dep.dep_vol || this_dep.ocean_parent \
+     || (this_dep.water_vol == this_dep.dep_vol && deps.at(this_dep.parent).water_vol == 0)) {
```

Comment:
We've discussed this bug in the past and I didn't realise the pull hadn't been completed. 
This allows depressions to fill in cases where they are exactly full, with an empty parent.